### ISSUE TITLE
Cannot build docs

### ISSUE THREAD

**Issue by cbrnr** [mne-tools/mne-python#12566]
I just tried to build the docs locally:

1. `pip install -e ".[doc]"`
2. `cd doc`
3. `make html-noplot`

This generates the following error:

```
sphinx-build -D plot_gallery=0 -b html -d _build/doctrees -nWT --keep-going . _build/html
Running Sphinx v7.2.6
Building documentation for MNE 1.8.0.dev9+gdef0503f2 (/Users/clemens/Projects/mne-python/mne/__init__.py)
Using pyvistaqt 3d backend.

making output directory... done
WARNING: optipng binaries not found, PNG images and thumbnails will not be optimized
Using Sphinx-Gallery to convert rst text blocks to markdown for .ipynb files.
checking bibtex cache... out of date
parsing bibtex file /Users/clemens/Projects/mne-python/doc/references.bib... parsed 222 entries
[autosummary] generating autosummary for: api/connectivity.rst, api/covariance.rst, api/creating_from_arrays.rst, api/datasets.rst, api/decoding.rst, api/events.rst, api/export.rst, api/file_io.rst, api/forward.rst, api/inverse.rst, ..., install/installers.rst, install/manual_install.rst, install/manual_install_python.rst, install/mne_c.rst, install/mne_tools_suite.rst, install/updating.rst, martinos.rst, mne_cpp.rst, old_versions/index.rst, overview/people.rst
[autosummary] generating autosummary for: /Users/clemens/Projects/mne-python/doc/generated/mne.AcqParserFIF.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.Annotations.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.BaseEpochs.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.BiHemiLabel.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.Covariance.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.Dipole.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.DipoleFixed.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.Epochs.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.EpochsArray.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.Evoked.rst, ..., /Users/clemens/Projects/mne-python/doc/generated/mne.write_events.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.write_evokeds.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.write_forward_solution.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.write_head_bem.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.write_label.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.write_labels_to_annot.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.write_proj.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.write_source_spaces.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.write_surface.rst, /Users/clemens/Projects/mne-python/doc/generated/mne.write_trans.rst
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://numpy.org/doc/stable/objects.inv...
loading intersphinx inventory from https://docs.scipy.org/doc/scipy/objects.inv...
loading intersphinx inventory from https://matplotlib.org/stable/objects.inv...
loading intersphinx inventory from https://scikit-learn.org/stable/objects.inv...
loading intersphinx inventory from https://numba.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://joblib.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://nipy.org/nibabel/objects.inv...
loading intersphinx inventory from http://nilearn.github.io/stable/objects.inv...
loading intersphinx inventory from https://nipy.org/nitime/objects.inv...
loading intersphinx inventory from https://pysurfer.github.io/objects.inv...
loading intersphinx inventory from https://mne.tools/mne-bids/stable/objects.inv...
loading intersphinx inventory from https://mne.tools/mne-connectivity/stable/objects.inv...
loading intersphinx inventory from https://mne.tools/mne-gui-addons/objects.inv...
loading intersphinx inventory from https://pandas.pydata.org/pandas-docs/stable/objects.inv...
loading intersphinx inventory from https://seaborn.pydata.org/objects.inv...
loading intersphinx inventory from https://www.statsmodels.org/dev/objects.inv...
loading intersphinx inventory from https://patsy.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://docs.pyvista.org/objects.inv...
loading intersphinx inventory from https://imageio.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://pierreablin.github.io/picard/objects.inv...
loading intersphinx inventory from https://eeglabio.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://docs.dipy.org/stable/objects.inv...
loading intersphinx inventory from https://pybv.readthedocs.io/en/latest/objects.inv...
loading intersphinx inventory from https://pyqtgraph.readthedocs.io/en/latest/objects.inv...
intersphinx inventory has moved: https://docs.pyvista.org/objects.inv -> https://docs.pyvista.org/version/stable/objects.inv
generating gallery...
generating gallery for auto_tutorials... 
/Users/clemens/.pyenv/versions/3.12.2/envs/mne/lib/python3.12/site-packages/sphinx_gallery/py_source_parser.py:101: DeprecationWarning: ast.Str is deprecated and will be removed in Python 3.14; use ast.Constant instead
  and isinstance(node.body[0].value, ast.Str)
/Users/clemens/.pyenv/versions/3.12.2/envs/mne/lib/python3.12/site-packages/sphinx_gallery/py_source_parser.py:113: DeprecationWarning: Attribute s is deprecated and will be removed in Python 3.14; use value instead
  if len(node.body[0].value.s) and node.body[0].value.s[0] == "\n":
generating gallery for auto_tutorials/intro... [100%] 70_report.py
generating gallery for auto_tutorials/io... [100%] 70_reading_eyetracking_data.py
generating gallery for auto_tutorials/raw... [100%] 40_visualize_raw.py
generating gallery for auto_tutorials/preprocessing... [100%] 90_eyetracking_data.py
generating gallery for auto_tutorials/epochs... [100%] 60_make_fixed_length_epochs.py
generating gallery for auto_tutorials/evoked... [100%] 40_whitened.py
generating gallery for auto_tutorials/time-freq... [100%] 50_ssvep.py
generating gallery for auto_tutorials/forward... [100%] 90_compute_covariance.py
generating gallery for auto_tutorials/inverse... [100%] 95_phantom_KIT.py
generating gallery for auto_tutorials/stats-sensor-space... [100%] 75_cluster_ftest_spatiotemporal.py
generating gallery for auto_tutorials/stats-source-space... [100%] 60_cluster_rmANOVA_spatiotemporal.py
generating gallery for auto_tutorials/machine-learning... [100%] 50_decoding.py
generating gallery for auto_tutorials/clinical... [100%] 60_sleep.py
generating gallery for auto_tutorials/simulation... [100%] 80_dics.py
generating gallery for auto_tutorials/visualization... [100%] 20_ui_events.py
generating gallery for auto_examples... 
generating gallery for auto_examples/io... [100%] read_xdf.py
generating gallery for auto_examples/simulation... [100%] source_simulator.py
generating gallery for auto_examples/preprocessing... [100%] xdawn_denoising.py
generating gallery for auto_examples/visualization... [100%] xhemi.py
generating gallery for auto_examples/time_frequency... [100%] time_frequency_simulated.py
generating gallery for auto_examples/stats... [100%] sensor_regression.py
generating gallery for auto_examples/decoding... [100%] ssd_spatial_filters.py
generating gallery for auto_examples/connectivity... 
generating gallery for auto_examples/forward... [100%] source_space_morphing.py
generating gallery for auto_examples/inverse... [100%] vector_mne_solution.py
generating gallery for auto_examples/datasets... [100%] spm_faces_dataset.py
generating MNE command help ... [ 43%] mne_freeview_bem_surfaces.py
Traceback (most recent call last):
  File "/Users/clemens/.pyenv/versions/3.12.2/envs/mne/lib/python3.12/site-packages/sphinx/events.py", line 97, in emit
    results.append(listener.handler(self.app, *args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/clemens/Projects/mne-python/doc/sphinxext/gen_commands.py", line 75, in generate_commands_rst
    module.run()
  File "/Users/clemens/Projects/mne-python/mne/commands/mne_freeview_bem_surfaces.py", line 88, in run
    subjects_dir = get_subjects_dir()
                   ^^^^^^^^^^^^^^^^^^
  File "/Users/clemens/Projects/mne-python/mne/utils/config.py", line 474, in get_subjects_dir
    subjects_dir = _check_fname(
                   ^^^^^^^^^^^^^
  File "<decorator-gen-0>", line 12, in _check_fname
  File "/Users/clemens/Projects/mne-python/mne/utils/check.py", line 280, in _check_fname
    raise FileNotFoundError(f'{name} does not exist: "{fname}"')
FileNotFoundError: subjects_dir does not exist: "/Users/clemens/Desktop/temp/freesurfer"

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/clemens/.pyenv/versions/3.12.2/envs/mne/lib/python3.12/site-packages/sphinx/cmd/build.py", line 293, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/clemens/.pyenv/versions/3.12.2/envs/mne/lib/python3.12/site-packages/sphinx/application.py", line 272, in __init__
    self._init_builder()
  File "/Users/clemens/.pyenv/versions/3.12.2/envs/mne/lib/python3.12/site-packages/sphinx/application.py", line 343, in _init_builder
    self.events.emit('builder-inited')
  File "/Users/clemens/.pyenv/versions/3.12.2/envs/mne/lib/python3.12/site-packages/sphinx/events.py", line 108, in emit
    raise ExtensionError(__("Handler %r for event %r threw an exception") %
sphinx.errors.ExtensionError: Handler <function generate_commands_rst at 0x332ecfa60> for event 'builder-inited' threw an exception (exception: subjects_dir does not exist: "/Users/clemens/Desktop/temp/freesurfer")

Extension error (gen_commands):
Handler <function generate_commands_rst at 0x332ecfa60> for event 'builder-inited' threw an exception (exception: subjects_dir does not exist: "/Users/clemens/Desktop/temp/freesurfer")
make: *** [html-noplot] Error 2
```

I hope I don't have to install Freesurfer just to build the docs?

**Comment by larsoner** [comment]
No need for FreeSurfer -- the error is `subjects_dir does not exist` so It wants a valid `subjects_dir`. In principle it shouldn't, but an easy workaround is to `mne.datasets.fetch_fsaverage` then set your SUBJECTS_DIR in your MNE config to point there.

Then we should also look into why it even tries to run the function, it shouldn't do that when in `--help` mode

**Comment by larsoner** [comment]
Try updating to latest `latexcodec`

**Comment by larsoner** [comment]
Hah actually latest `latexcodec` seems to have the problem. 2.something didn't. Just add the warning to `doc/conf.py` warnings list

https://github.com/mne-tools/mne-python/blob/52d9a9b5a03503d00439c7ba999799d8dd4ac030/doc/conf.py#L1364

**Comment by cbrnr** [comment]
> Just add the warning to `doc/conf.py` warnings list

OK, that worked, thanks. How would you like to proceed in order to fix this issue/these issues?

**Comment by larsoner** [comment]
> I've created the temp folder for now, but I really shouldn't have to do that. I didn't want to download the dataset and change the config, but just creating the folder was enough (I think).

In your case I think it was essentially that your mne-python config was invalid -- it pointed to a directory that did not exist. So the error might have been a bit cryptic but ultimately perhaps helpful in that it indicated something was indeed wrong. I'd suggest just to delete the config value directly or use `mne.set_config("SUBJECTS_DIR", None)` to have MNE do it for you. I'll open a PR to improve the message

> How would you like to proceed in order to fix this issue/these issues?

Any PR could include this new ignore in the ignore lines. I can put it in my config-checking one.

**Comment by cbrnr** [comment]
> In your case I think it was essentially that your mne-python config was invalid -- it pointed to a directory that did not exist.

Correct, thanks for improving the error message! And yes, please go ahead and include the new ignore.

**Comment by cbrnr** [comment]
But what about the real underlying issue that the directory should not be required in the first place?

**Comment by cbrnr** [comment]
Fixed by #12567.
